APIKEY auth added for all v1 routes

SadiqOnGithub opened the pull request:
![image](https://github.com/user-attachments/assets/fe21a03d-4f39-41a1-b9d5-73fc971a8aa3)

export API_KEY=123

SadiqOnGithub committed:
APIKEY auth added for all v1 routes

codebanesr commented:
@SadiqOnGithub read from kubernetes secret, also add an entry in makefile to update the secret and redeploy. 

research pre / post deployment, things to watch out for etc

SadiqOnGithub committed:
feat: Read config from Kubernetes secret and add deployment checklist

This commit addresses maintainer feedback by:
- Modifying internal/config/config.go to read API_KEY and SANDBOX_TIMEOUT_MINUTES from files mounted from a Kubernetes secret.
- Updating kubernetes/manifests/deployment.yaml to mount the secret.
- Creating kubernetes/manifests/secret.yaml with default secret values.
- Updating kubernetes/manifests/kustomization.yaml to include the new secret.
- Adding a 'update-secret-and-redeploy' target to the Makefile for easier secret updates and redeployments.
- Creating deployment-checklist.md summarizing pre and post-deployment best practices for Kubernetes.
